pkg/controller/instance/controller_status.go

```diff
@@ -16,34 +16,150 @@ package instance
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 
 	"github.com/kro-run/kro/api/v1alpha1"
+	"github.com/kro-run/kro/pkg/apis"
 	"github.com/kro-run/kro/pkg/requeue"
 )
 
-func createCondition(conditionType v1alpha1.ConditionType, status corev1.ConditionStatus, reason, message string, generation int64) map[string]interface{} {
-	return map[string]interface{}{
-		"type":               string(conditionType),
-		"status":             string(status),
-		"reason":             reason,
-		"message":            message,
-		"lastTransitionTime": time.Now().Format(time.RFC3339),
-		"observedGeneration": generation,
+const (
+	Ready           = "Ready"
+	InstanceManaged = "InstanceManaged"
+	GraphResolved   = "GraphResolved"
+	ResourcesReady  = "ResourcesReady"
+)
+
+var instanceConditionTypes = apis.NewReadyConditions(InstanceManaged, GraphResolved, ResourcesReady)
+
+// unstructuredConditionAdapter adapts an unstructured.Unstructured to implement apis.Object
+type unstructuredConditionAdapter struct {
+	*unstructured.Unstructured
+}
+
+// GetConditions implements apis.Object interface
+func (u *unstructuredConditionAdapter) GetConditions() []v1alpha1.Condition {
+	if conditions, found, err := unstructured.NestedSlice(u.Object, "status", "conditions"); err == nil && found {
+		// Marshal the conditions slice to JSON and then unmarshal to []v1alpha1.Condition
+		conditionsJSON, err := json.Marshal(conditions)
+		if err != nil {
+			return []v1alpha1.Condition{}
+		}
+
+		var result []v1alpha1.Condition
+		if err := json.Unmarshal(conditionsJSON, &result); err != nil {
+			return []v1alpha1.Condition{}
+		}
+		return result
 	}
+	return []v1alpha1.Condition{}
+}
+
+// SetConditions implements apis.Object interface
+func (u *unstructuredConditionAdapter) SetConditions(conditions []v1alpha1.Condition) {
+	// Marshal the conditions to JSON and then unmarshal to interface{} slice
+	conditionsJSON, err := json.Marshal(conditions)
+	if err != nil {
+		return // Fail silently - could log this in the future
+	}
+
+	var conditionsInterface []interface{}
+	if err := json.Unmarshal(conditionsJSON, &conditionsInterface); err != nil {
+		return // Fail silently - could log this in the future
+	}
+
+	if err := unstructured.SetNestedSlice(u.Object, conditionsInterface, "status", "conditions"); err != nil {
+		return // Fail silently - could log this in the future
+	}
+}
+
+// wrapInstance creates an adapter that allows unstructured instances to work with condition management
+func wrapInstance(instance *unstructured.Unstructured) *unstructuredConditionAdapter {
+	return &unstructuredConditionAdapter{Unstructured: instance}
+}
+
+// NewConditionsMarkerFor creates a marker to manage conditions and sub-conditions for instances.
+//
+// ```
+// Ready
+//	├─ InstanceManaged - Instance finalizers and labels are properly set
+//	├─ GraphResolved - Runtime graph created and all resources resolved
+//	└─ ResourcesReady - All resources are created and ready
+//
+// ```
```

Comment:
```suggestion
// NewConditionsMarkerFor creates a marker to manage conditions and sub-conditions for instances.
// ```
// Ready
//	├─ InstanceManaged - Instance finalizers and labels are properly set
//	├─ GraphResolved - Runtime graph created and all resources resolved
//	└─ ResourcesReady - All resources are created and ready
// ```
```